packages/extension-commerce-bm-seo/src/utils/route-match-utils.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {matchPath as matchPathReactRouter} from 'react-router-dom'
+import {RouteProps} from '@salesforce/pwa-kit-extension-sdk/types'
+
+/**
+ * Checks whether the given URL path matches a predefined route defined in the application's routes config.
```

Comment:
Lets be super explicit here on what we doing here. We are creating an amped up version of react routers matchPath method. 

So how about we simply say that.

"This is an enhanced version of matchPath that allows you to match to multiple routes as well as allowing you to filter out wildcard routes."